pkg/operator/sync.go

```diff
@@ -101,6 +102,24 @@ func (optr *Operator) syncAll(config *OperatorConfig) error {
 
 func (optr *Operator) syncClusterAPIController(config *OperatorConfig) error {
 	controllersDeployment := newDeployment(config, nil)
+
+	// we watch some resources so that our deployment will redeploy without explicitly and carefully ordered resource creation
+	inputHashes, err := resourcehash.MultipleObjectHashStringMapForObjectReferences(
```

Comment:
so is this logic to rollout the deployment replicas if the configMap changes?
If so why is that necessary?
shouldn't this be updated automatically https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#mounted-configmaps-are-updated-automatically

Reply:
can we put this logic in a unit tested func that we just invoke here?

Reply:
So i created the `forceSyncDeployment` function that takes in inputHashes and then applies that to the deployment, and added a unit test for that.

generation of the inputhashes is function from library-go, so personally it didn't make sense to test it's behavior again as part of unit testing here.

Reply:
ok, thanks!
personally I see `// forceSyncDeployment uses inputHash map of external dependencies to force new generation of the deployment
// triggering the Kubernetes rollout as defined when the inputHash changes.`
as a circumstantial consequence and nice to have a comment but I would name the function for what it actually does `ensureDependecyAnnotations` or something like that